src/metabase/driver/postgres.clj

```diff
@@ -292,28 +292,32 @@
 (def ^:private ssl-params
   "Params to include in the JDBC connection spec for an SSL connection."
   {:ssl        true
-   :sslmode    "require"
-   :sslfactory "org.postgresql.ssl.NonValidatingFactory"})
+   :sslmode    "require"})
 
 (def ^:private disable-ssl-params
   "Params to include in the JDBC connection spec to disable SSL."
   {:sslmode "disable"})
 
 (defmethod sql-jdbc.conn/connection-details->spec :postgres
   [_ {ssl? :ssl, :as details-map}]
-  (-> details-map
-      (update :port (fn [port]
-                      (if (string? port)
-                        (Integer/parseInt port)
-                        port)))
-      ;; remove :ssl in case it's false; DB will still try (& fail) to connect if the key is there
-      (dissoc :ssl)
-      (merge (if ssl?
-               ssl-params
-               disable-ssl-params))
-      (set/rename-keys {:dbname :db})
-      db.spec/postgres
-      (sql-jdbc.common/handle-additional-options details-map)))
+  (let [props (-> details-map
+                  (update :port (fn [port]
+                                  (if (string? port)
+                                    (Integer/parseInt port)
+                                    port)))
+                  ;; remove :ssl in case it's false; DB will still try (& fail) to connect if the key is there
+                  (dissoc :ssl))
+        ;; this happens via ->> so that the users props will override the ssl-params stuff.
+        ;; if the user has specified a sslmode, it must always take precedence over our default.
+        props (->> props
+                   (merge (if ssl?
+                            ssl-params
+                            disable-ssl-params)))
+        props (-> props
+                  (set/rename-keys {:dbname :db})
+                  db.spec/postgres
+                  (sql-jdbc.common/handle-additional-options details-map))]
```

Comment:
one thing, @camsaul - is there a better way to do this? `->` works for everything *but* the `merge` I'm doing, where I specifically want the user-specified `props` to take precedence.

Reply:
A couple ideas: you could make `(merge ... props)` the first arg in to the `->` form e.g.

```clj
(-> (merge ... props)
    (set/rename-keys ...)
    ...)
```

2. You could use `as->` e.g.

```clj
(as-> props props
  (merge ... props)
  (set/rename-keys props {:dbname :db})
  ...)
```

There's some disagreement about the true purpose of `as->` so some people would tell you to do something like this instead:

```clj
(-> props
    (as-> props (merge props {...}))
    db.spec/postgres
    ...)
```

But I think nesting threading macros like that makes things too confusing